backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/document/model/dsl/impl/ElasticsearchRootIndexSchemaCollectorImpl.java

```diff
@@ -9,12 +9,14 @@
 import org.hibernate.search.v6poc.backend.document.model.dsl.spi.IndexSchemaNestingContext;
 import org.hibernate.search.v6poc.backend.elasticsearch.document.model.dsl.ElasticsearchIndexSchemaElement;
 import org.hibernate.search.v6poc.backend.elasticsearch.document.model.impl.ElasticsearchIndexSchemaNodeCollector;
+import org.hibernate.search.v6poc.backend.elasticsearch.document.model.impl.ElasticsearchRootIndexSchemaContributor;
 import org.hibernate.search.v6poc.backend.elasticsearch.document.model.impl.esnative.RootTypeMapping;
 import org.hibernate.search.v6poc.backend.elasticsearch.document.model.impl.esnative.RoutingType;
 import org.hibernate.search.v6poc.backend.elasticsearch.multitenancy.impl.MultiTenancyStrategy;
 
 public class ElasticsearchRootIndexSchemaCollectorImpl
-		extends AbstractElasticsearchIndexSchemaCollector<IndexSchemaRootTypeNodeBuilder> {
+		extends AbstractElasticsearchIndexSchemaCollector<IndexSchemaRootTypeNodeBuilder> implements
+		ElasticsearchRootIndexSchemaContributor {
```

Comment:
I'm a bit puzzled with the class being named `Collector` and the interface being named `Contributor`. Is it expected?

Reply:
Right... It's a pre-existing thing, really. The collector already had a `contribute` method before.

What is going on is this: the object collects data from the mapper through the `IndexSchemaCollector` interface, so it's a collector from the point of view of the mapper.
But the collected data must go somewhere. So this object will also contribute to the backend-specific model. Thus it is a contributor from the point of view of the backend.

Note that the same goes with field contexts: from the point of view of the mapper, they are mostly collectors, but from the point of view of the backend, they implement `ElasticsearchIndexSchemaNodeContributor`.

It's a bit convoluted, but I think makes sense. And really, there's not much choice if we want to prevent cyclic dependencies here.

Maybe we could rename the contributor interfaces to `*Model*Contributor` to make it more explicit? But then for consistency I would have to rename `ElasticsearchIndexSchemaObjectNode` and such to `*ModelNode`, and if I remember correctly you were against that.

Reply:
OK, then.